src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,751 @@
+
+  /**
+   * Called when the status of the drawer changes.
+   */
+  onDrawerStateChanged?: (
+    newState: DrawerState,
+    drawerWillShow: boolean
+  ) => void;
+
+  /**
+   * Type of animation that will play when opening the drawer.
+   */
+  drawerType?: DrawerType;
+
+  /**
+   * Defines how far from the edge of the content view the gesture should
+   * activate.
+   */
+  edgeWidth?: number;
+
+  /**
+   * Minimal distance to swipe before the drawable starts moving.
+   */
+  minSwipeDistance?: number;
+
+  /**
+   * When set to true Drawer component will use
+   * {@link https://reactnative.dev/docs/statusbar StatusBar} API to hide the OS
+   * status bar whenever the drawer is pulled or when its in an "open" state.
+   */
+  hideStatusBar?: boolean;
+
+  /**
+   * @default 'slide'
+   *
+   * Can be used when hideStatusBar is set to true and will select the animation
+   * used for hiding/showing the status bar. See
+   * {@link https://reactnative.dev/docs/statusbar StatusBar} documentation for
+   * more details
+   */
+  statusBarAnimation?: StatusBarAnimation;
+
+  /**
+   * @default 'rgba(0, 0, 0, 0.7)'
+   *
+   * Color of the background overlay.
+   * Animated from `0%` to `100%` as the drawer opens.
+   */
+  overlayColor?: string;
+
+  /**
+   * Style wrapping the content.
+   */
+  contentContainerStyle?: StyleProp<ViewStyle>;
+
+  /**
+   * Style wrapping the drawer.
+   */
+  drawerContainerStyle?: StyleProp<ViewStyle>;
+
+  /**
+   * Enables two-finger gestures on supported devices, for example iPads with
+   * trackpads. If not enabled the gesture will require click + drag, with
+   * `enableTrackpadTwoFingerGesture` swiping with two fingers will also trigger
+   * the gesture.
+   */
+  enableTrackpadTwoFingerGesture?: boolean;
+
+  onDrawerSlide?: (position: number) => void;
+
+  // Implicit `children` prop has been removed in @types/react^18.0.
+  /**
+   * Elements that will be rendered inside the content view.
+   */
+  children?: ReactNode | ((openValue?: SharedValue<number>) => ReactNode);
+
+  /**
+   * @default 'none'
+   * Defines which userSelect property should be used.
+   * Values: 'none'|'text'|'auto'
+   */
+  userSelect?: UserSelect;
+
+  /**
+   * @default 'auto'
+   * Defines which cursor property should be used when gesture activates.
+   * Values: see CSS cursor values
+   */
+  activeCursor?: ActiveCursor;
+
+  /**
+   * @default 'MouseButton.LEFT'
+   * Allows to choose which mouse button should underlying pan handler react to.
+   */
+  mouseButton?: MouseButton;
+
+  /**
+   * @default 'false if MouseButton.RIGHT is specified'
+   * Allows to enable/disable context menu.
+   */
+  enableContextMenu?: boolean;
+}
+
+export type DrawerMovementOption = {
+  initialVelocity?: number;
+  animationSpeed?: number;
+};
+
+export interface DrawerLayoutMethods {
+  openDrawer: (options?: DrawerMovementOption) => void;
+  closeDrawer: (options?: DrawerMovementOption) => void;
+}
+
+const defaultProps = {
```

Comment:
Typing `defaultProps` as `Partial<DrawerLayoutProps>` stops it from being automatically typed as an object of  literal props, meaning it can no longer be used as a setter of default values, since nearly all it's props are `| undefined` now.
Not sure if there's a way around it.

![image](https://github.com/user-attachments/assets/525e242e-2326-46e7-ba37-c47e774f774c)
![image](https://github.com/user-attachments/assets/716aa990-5b91-4695-b134-76e43ee6ef03)

Reply:
After some further investigation it looks like both our prettier and babel versions are too dated to use this feature. Changing them would require a separate PR。Is it ok if we ignore this issue for now, and I'll fix it in a separate PR?

Reply:
Yep, makes sense